### PR TITLE
Avoid NaNs in Spherical and Spins Transforms

### PR DESCRIPTION
**Pull request by SSL32081**
This PR attempts to address part of issue #204, specifically, the `nan` occurs at particular parameter values. 

The whole concept is to redefine the `arccos` and `arctan2` functions at the problematic points so that the gradient has a natural extension at those points. 

Explicitly, they are:
```python
theta = jnp.where(
    (jnp.abs(x) < EPS) & (jnp.abs(y) < EPS),
    jnp.arccos(jnp.sign(z)),
    jnp.arccos(z / jnp.sqrt(x ** 2 + y ** 2 + z ** 2)),
)

phi = jnp.where(
    (jnp.abs(x) < EPS) & (jnp.abs(y) < EPS),
    default_value * jnp.ones_like(x),
    jnp.atan2(y, x),
)
```

## Summary by CodeRabbit

- **New Features**
  - Improved numerical stability for angle calculations, reducing the risk of undefined values in edge cases.
- **Refactor**
  - Unified and streamlined angle computations across the application for consistency.
- **Style**
  - Enhanced code readability with minor formatting and spacing adjustments.

### CODE REVIEW

**Comment by SSL32081** [comment]
@coderabbitai review

**Comment by SSL32081** [comment]
@kazewong I have updated the docstring style and fixed the merge conflict, it is ready for review.

**Review by kazewong — COMMENTED** [review]
Approved

**Review by kazewong — APPROVED** [review]

**Review by kazewong — APPROVED** [review]
